.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -31,6 +31,54 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  get-test-directories:
+    if: "!contains(fromJSON(inputs.skip).jobs, 'opal-validation')"
+    name: Get Test Directories For OPAL Validation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install Python
+        uses: actions/setup-python@v2
+      - name: Build matrix
+        id: matrix
+        shell: bash
+        run: |
+          DIRS=$(python -c "import json; import glob; print(json.dumps([x.replace('/main.tf', '') for x in glob.glob('.tests/**/main.tf', recursive=True)]))")
```

Comment:
Can you just use `ls` instead? That would avoid installing python. You can assume any directory under ./tests is intended as a test.

Reply:
the tricky part was getting bash to format the results of `ls` to a list-friendly string, like `["test1","test2"]`, but eventually i think i got it.